pkg/scheduler/framework/plugins/podtopologyspread/filtering_test.go

```diff
@@ -1450,12 +1534,41 @@ func TestSingleConstraint(t *testing.T) {
 				"node-y": framework.Unschedulable,
 			},
 		},
+		{
+			name: "pods spread across nodes as 2/2/1, maxSkew is 2, and the number of domains < minDomains, then the third node fits",
+			pod: st.MakePod().Name("p").Label("foo", "").SpreadConstraint(
+				2,
+				"node",
```

Comment:
Can you add another test to cover "zone"? as some bug may hide in the non-node level. Something like:

- node1, node2 belongs to zoneA
- node3, node4 belongs to zoneB
- existing pods spread as 1/1/0/1
- maxSkew=1, and minDomains=3
- none of the nodes can accommodate incoming pod

Also let's add a test to cover "minDomains >= # of present zones"